### PR TITLE
Added delete and update of questions and answers + a lot more

### PR DESCRIPTION
**Pull request by TobiasFriberg**
* Updated to sass.. because, you know.. less i less, sass is more, rite? :)
* Updated design-language to Material Design, looks totally awesome!
* Added profile picture to user.
* Added a few necessary components (like dialog and stuff).
* Made UD in CRUD work proper.
* Reworked components that was a bit wonky and sketchy, we want the best UX, rite?
* Added meta tag for language, made Markdown work proper (not even my fault).
* Removed title from answer.
* Updated 404 with totally awesome art work!
* Added artwork for 401.

Overall this is a solid piece, Don't let the size scare you off.

### CODE REVIEW

File: spec/src/test/java/impl/UserAnswerResourceTest.java [hunk 1]
```diff
@@ -130,9 +127,6 @@ public void shouldBeAbleToDeleteAnswer() {
         // then only the answer we want to delete should be deleted
         List<Answer> remainingAnswers = userAnswerResource.getAnswers(createdUser.getId(), question.getId()).toBlocking().single();
         assertThat(remainingAnswers.size()).isEqualTo(1);
-        assertThat(remainingAnswers)
```
Comment (jepp3) [inline spec/src/test/java/impl/UserAnswerResourceTest.java]: shortlived , but i will always remember .. :)

File: ui/src/components/questions/post.js [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+class Post extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            votes: 0,
+            isDeleteDialogOpen: false,
+            isEditDialogOpen: false,
+            currentBody: '',
+            postingToServer: false
```
Comment (jepp3) [inline ui/src/components/questions/post.js]: maybe a better name?

File: ui/src/components/questions/post.js [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+
+    componentDidMount() {
+        this.setState({
+            votes: this.props.votes
+        });
+    }
+
+    getClasses() {
+        let classes = `question-post ${this.props.className}`;
+        if (this.state.votes < -3) {
```
Comment (jepp3) [inline ui/src/components/questions/post.js]: i think you can flip it around 
```javascript
if(answered) {
 return `${classes} accepted`
}
if(bad..voting..) {
 return `${classes} faded`
}
return classes;
```

File: ui/src/components/questions/post.js [hunk 4]
```diff
@@ -0,0 +1,277 @@
...
+        });
+    }
+
+    getTime() {
+        return moment(this.props.created).fromNow();
+    }
+
+    renderAnswered() {
+        return (
+            <Certificate active={this.props.answered} />
```
Comment (jepp3) [inline ui/src/components/questions/post.js]: maybe a more precise name?

File: ui/src/components/questions/post.js [hunk 5]
```diff
@@ -0,0 +1,277 @@
...
+        return t`Delete question`;
+    }
+
+    getDialogBody() {
+        let prefix = <b>{this.props.title}</b>;
+        if (!this.props.questionId) {
+            prefix = t`The answer`;
+        }
+
+        return <>{prefix} {t`will be deleted from Rocket fuel.`}</>;
```
Comment (jepp3) [inline ui/src/components/questions/post.js]: is it supposed to be empty tag here?

File: ui/src/components/questions/post.js [hunk 6]
```diff
@@ -0,0 +1,277 @@
...
+        }
+        this.props.history.push(`/create/thread/${this.props.questionId}`);
+    }
+
+    saveAnswer() {
+        if (!this.props.answerId) {
+            return;
+        }
+        this.setState({
+            postingToServer: true
```
Comment (jepp3) [inline ui/src/components/questions/post.js]: same for this, maybe go over all postingToServer

File: ui/src/components/questions/post.js [hunk 7]
```diff
@@ -0,0 +1,277 @@
...
+                <Button color="secondary" text small onClick={this.openDeleteDialog.bind(this)}>
+                    <i className="fa fa-trash" /> {t`Delete`}
+                </Button>
+            </div>
+        );
+    }
+
+    render() {
+        return (
+            <div className={this.getClasses()}>
```
Comment (jepp3) [inline ui/src/components/questions/post.js]: Do we think that we can reduce the size of this render into smaller parts? or is it easier to read this way?

File: ui/src/components/questions/questionrow.js [hunk 8]
```diff
@@ -0,0 +1,198 @@
...
+
+    getTime() {
+        return moment(this.props.question.createdAt).fromNow();
+    }
+
+    getState() {
+        if (this.props.question.answerAccepted) {
+            return 'answered';
+        }
+
```
Comment (jepp3) [inline ui/src/components/questions/questionrow.js]: space :)

File: ui/src/index.html [hunk 9]
```diff
@@ -1,6 +1,7 @@
-<html>
-
+<html lang="en">
```
Comment (jepp3) [inline ui/src/index.html]: should we have a <!doctype html> as well?

File: ui/src/models/user.js [hunk 10]
```diff
@@ -28,3 +28,61 @@ export function signIn(token) {
...
+}
+
+export function updateQuestion(questionId, body) {
+    const options = {
+        url: `/api/users/me/questions/${questionId}`,
+        method: 'PUT',
+        body: body
+    };
+
+    return ApiFetch(options).then((response) => {
```
Comment (jepp3) [inline ui/src/models/user.js]: quite repetitive, maybe find a way to reuse the default if statements if(error) then ..

File: ui/src/components/questions/questionrow.js [hunk 11]
```diff
@@ -0,0 +1,198 @@
...
+        if (!this.state.isDropdownOpen) {
+            return;
+        }
+        this.setState({
+            isDropdownOpen: false
+        });
+    }
+
+    editQuestion() {
+        this.props.history.push(`/create/thread/${this.props.question.id}`);
```
Comment (jepp3) [inline ui/src/components/questions/questionrow.js]: thread? :)

File: ui/src/components/questions/post.js [hunk 12]
```diff
@@ -0,0 +1,277 @@
...
+
+    editPost() {
+        if (!this.props.questionId) {
+            this.setState({
+                answer: this.props.body,
+                isEditDialogOpen: true
+            });
+            return;
+        }
+        this.props.history.push(`/create/thread/${this.props.questionId}`);
```
Comment (jepp3) [inline ui/src/components/questions/post.js]: We need to use the same terminology everywhere,. question i think is what we are creating here?

File: ui/src/components/questions/post.js (hunk 2 above)
Comment (TobiasFriberg) [reply ui/src/components/questions/post.js]: what's a better name when interacting with the server?

File: ui/src/components/questions/post.js (hunk 3 above)
Comment (TobiasFriberg) [reply ui/src/components/questions/post.js]: Fixed

File: ui/src/components/questions/post.js (hunk 4 above)
Comment (TobiasFriberg) [reply ui/src/components/questions/post.js]: It's a certificate symbol, so.. it's pretty accurate I would say. What did you have in mind?

File: ui/src/components/questions/post.js (hunk 5 above)
Comment (TobiasFriberg) [reply ui/src/components/questions/post.js]: Yes, it's just like <Fragment> but without the word Fragment.

File: ui/src/components/questions/post.js (hunk 6 above)
Comment (TobiasFriberg) [reply ui/src/components/questions/post.js]: Hook me up with a better name and I'll change it for you. Altho, I do think the name says pretty much what it does.

File: ui/src/components/questions/post.js (hunk 7 above)
Comment (TobiasFriberg) [reply ui/src/components/questions/post.js]: Fixed

File: ui/src/components/questions/questionrow.js (hunk 8 above)
Comment (TobiasFriberg) [reply ui/src/components/questions/questionrow.js]: Ok, now you just grasping for straws..

File: ui/src/index.html (hunk 9 above)
Comment (TobiasFriberg) [reply ui/src/index.html]: Yes, ofc.. fixed.

File: ui/src/models/user.js (hunk 10 above)
Comment (TobiasFriberg) [reply ui/src/models/user.js]: Fixed

File: ui/src/components/questions/questionrow.js (hunk 11 above)
Comment (TobiasFriberg) [reply ui/src/components/questions/questionrow.js]: Fixed all.

thread -> question

File: ui/src/components/questions/post.js (hunk 12 above)
Comment (TobiasFriberg) [reply ui/src/components/questions/post.js]: fixed

File: impl/src/main/java/dao/AnswerDao.java [hunk 13]
```diff
@@ -29,7 +28,7 @@
 
     @Query(
         SELECT_ANSWER +
-            "\"user\".\"name\" AS created_by " +
+            "\"user\".\"name\" AS created_by, \"user\".picture " +
```
Comment (rassmate) [inline impl/src/main/java/dao/AnswerDao.java]: Är det för att det är reserverade ord som gör att vi måste dubbelfnutta user?
Sen kanske du kunde ta bort dubbelfnuttningen på name här då om det inte behövs. Det verkar ju inte behövas på rad 40 iaf.

Comment (flowertwig) [reply impl/src/main/java/dao/AnswerDao.java]: User is reserved but we could use an alias, e.g. INNER JOIN \"user\" u, to minimise fnutts. :)

Comment (jepp3) [reply impl/src/main/java/dao/AnswerDao.java]: as well , we should rename the tables to plural. But thats another issue

**Pull request merged**